src/odemis/driver/test/orsay_test.py

```diff
@@ -712,104 +655,153 @@ def test_errorstate(self):
         sleep(1)
         self.assertEqual(self.gis.state.value, model.ST_RUNNING)
 
-    def test_updatePosition(self):
+        self.gis._gis.ErrorState.Actual = init_state  # return to value from before test
+
+    def test_updatePositionArm(self):
         """
-        Check that the position VA is updated correctly and an exception is raised when the wrong parameter is passed
+        Check that the "arm" part of the position VA is updated correctly and an exception is raised when the wrong
+        parameter is passed
         """
         with self.assertRaises(ValueError):
             self.gis._updatePosition(self.datamodel.HybridPlatform.Cancel)
 
         if not TEST_NOHW == "sim":
             self.skipTest("TEST_NOHW is not set to sim, data isn't copied from Target to Actual outside of simulation.")
 
-        self.gis._gis.PositionState.Target = orsay.STR_WORK
+        init_pos = self.gis._positionPar.Target
+
+        self.gis._positionPar.Target = orsay.STR_WORK
         sleep(1)
         self.assertTrue(self.gis.position.value["arm"])
 
-        self.gis._gis.PositionState.Target = "MOVING"
+        self.gis._positionPar.Target = "MOVING"
         sleep(1)
         self.assertFalse(self.gis.position.value["arm"])
 
-        self.gis._gis.PositionState.Target = orsay.STR_PARK
+        self.gis._positionPar.Target = orsay.STR_PARK
         sleep(1)
         self.assertFalse(self.gis.position.value["arm"])
 
-    def test_updateInjectingGas(self):
+        self.gis._positionPar.Target = init_pos  # return to value from before test
+
+    def test_updatePositionReservoir(self):
         """
-        Check that the injectingGas VA is updated correctly and an exception is raised when the wrong parameter is passed
+        Check that the "reservoir" part of the position VA is updated correctly
         """
-        with self.assertRaises(ValueError):
-            self.gis._updateInjectingGas(self.datamodel.HybridPlatform.Cancel)
-
         if not TEST_NOHW == "sim":
             self.skipTest("TEST_NOHW is not set to sim, data isn't copied from Target to Actual outside of simulation.")
```

Comment:
I don't think that's the reason you skip this test with the real hardware. I guess the test case would actually pass. But it's not a great idea to open the GIS randomly during tests, right?

Reply:
No, this is actually the reason and the testcase would fail on the real system. This holds for test_updatePositionReservoir and test_updatePositionArm. They would fail outside of simulation, because I'm writing "MOVING" to the Target, which is a value you should in reality never write to Target. I'm doing so here to test what the position VA does when the Actual has value "MOVING". These two test cases are only there to check that the VA has the correct value for every possible value of the Orsay parameter Actual values. Testing whether moving the arm and opening/closing the reservoir work as they should is done by test_moveAbs. I've clarified on this in the docstring of the two testcases.